quic/s2n-quic-transport/src/space/application.rs

```diff
@@ -147,7 +158,13 @@ impl<Config: endpoint::Config> ApplicationSpace<Config> {
     ) -> Result<(transmission::Outcome, EncoderBuffer<'a>), PacketEncodingError<'a>> {
         let mut packet_number = self.tx_packet_numbers.next();
 
+        // This function can return early and not transmit a packet for various reasons
+        // enumerated in PacketEncodingError. Record the intent to skip (de-duping, if
+        // necessary) and record only if a packet is transmitted.
+        let mut skipped_pn = SkippedPacketNumber::default();
```

Comment:
nit: I'd prefer just writing `packet_number` instead of `pn`. Abbreviations make the code less approachable IMO